src/Analysis/Ast/Impl/Modules/Resolution/MainModuleResolution.cs

```diff
@@ -187,7 +194,8 @@ public IPythonModule GetSpecializedModule(string name)
             var interpreterPaths = await GetSearchPathsAsync(cancellationToken);
             addedRoots.UnionWith(PathResolver.SetInterpreterSearchPaths(interpreterPaths));
 
-            addedRoots.UnionWith(SetUserSearchPaths(_interpreter.Configuration.SearchPaths));
+            var userSearchPaths = _interpreter.Configuration.SearchPaths.Except(interpreterPaths, StringExtensions.PathsStringComparer);
+            addedRoots.UnionWith(SetUserSearchPaths(userSearchPaths));
```

Comment:
I know this was here before you modified it, but does the use of a `HashSet` here mean that the path resolver is going to be nondeterministically choosing how import paths are resolved? From what I can tell, `ReloadModulePaths` iterates over `addedRoots`, calling `AddModulePath` on each, which then adds children to the path resolver internally in a specific order.

Reply:
Hmm, maybe that's not how it actually works, given this does `PathResolver.SetRoot`, `PathResolver.SetInterpreterSearchPaths`, then `PathResolver.SetUserSearchPaths` in a row before `ReloadModulesPaths`, so I probably just misunderstood what `ReloadModulePaths` did.

Reply:
@AlexanderSher 	added this set in #641 so I'm probably wrong about it all.